core/src/io/anuke/mindustry/world/Block.java

```diff
@@ -245,6 +245,38 @@ public void drawLayer(Tile tile){
     public void drawLayer2(Tile tile){
     }
 
+    public void drawBottleneck(Tile tile){
+        if(tile.entity == null) return;
+
+        switch(tile.block.category){
+            case production:
+            case crafting:
+            case power:
+            case units:
+                if(tile.block.hasItems || tile.block.hasLiquids || tile.block.hasPower){
+                    // bottom right center x/y
+                    float brcx = tile.drawx() + (size * tilesize / 2f) - (tilesize / 2f);
+                    float brcy = tile.drawy() - (size * tilesize / 2f) + (tilesize / 2f);
+                    Draw.color(bottleneckState(tile).color);
+                    Draw.rect(bottleneck, brcx, brcy);
+                    Draw.color();
+                }
+        }
+    }
+
+    public Bottleneck bottleneckState(Tile tile){
```

Comment:
I would put this method in `ConsumeModule`, since it has little to do with blocks themselves.